source/extensions/dynamic_modules/abi.h

```diff
@@ -2726,6 +2906,249 @@ bool envoy_dynamic_module_callback_network_filter_is_ssl(
 void envoy_dynamic_module_callback_network_filter_disable_close(
     envoy_dynamic_module_type_network_filter_envoy_ptr filter_envoy_ptr, bool disabled);
 
+// =============================================================================
+// ----------------------------- Listener Filter Callbacks ---------------------
+// =============================================================================
+
+// -----------------------------------------------------------------------------
+// Buffer Operations
+// -----------------------------------------------------------------------------
+
+/**
+ * envoy_dynamic_module_callback_listener_filter_get_buffer_slice is called by the module to
+ * get the current data buffer as a single slice. This is only valid during the
+ * envoy_dynamic_module_on_listener_filter_on_data callback.
+ *
+ * @param filter_envoy_ptr is the pointer to the DynamicModuleListenerFilter object.
+ * @param data_ptr_out is the output pointer to the buffer data.
+ * @param data_length_out is the output pointer to the length of the data.
+ * @return true if the buffer is available, false otherwise.
+ *
+ * The returned data is owned by Envoy and valid until the end of the callback.
+ */
+bool envoy_dynamic_module_callback_listener_filter_get_buffer_slice(
+    envoy_dynamic_module_type_listener_filter_envoy_ptr filter_envoy_ptr,
+    envoy_dynamic_module_type_buffer_envoy_ptr* data_ptr_out, size_t* data_length_out);
+
+/**
+ * envoy_dynamic_module_callback_listener_filter_drain_buffer is called by the module to drain
+ * bytes from the beginning of the buffer.
+ *
+ * @param filter_envoy_ptr is the pointer to the DynamicModuleListenerFilter object.
+ * @param length is the number of bytes to drain from the beginning of the buffer.
+ * @return true if the drain was successful, false otherwise.
+ */
+bool envoy_dynamic_module_callback_listener_filter_drain_buffer(
+    envoy_dynamic_module_type_listener_filter_envoy_ptr filter_envoy_ptr, size_t length);
+
+// -----------------------------------------------------------------------------
+// Socket Property Setters (Protocol Detection)
+// -----------------------------------------------------------------------------
+
+/**
+ * envoy_dynamic_module_callback_listener_filter_set_detected_transport_protocol is called by the
+ * module to set the detected transport protocol (e.g., "tls", "raw_buffer").
+ *
+ * @param filter_envoy_ptr is the pointer to the DynamicModuleListenerFilter object.
+ * @param protocol is the pointer to the protocol string.
+ * @param length is the length of the protocol string.
+ */
+void envoy_dynamic_module_callback_listener_filter_set_detected_transport_protocol(
+    envoy_dynamic_module_type_listener_filter_envoy_ptr filter_envoy_ptr,
+    envoy_dynamic_module_type_const_buffer_envoy_ptr protocol, size_t length);
```

Comment:
Could you check the point here, I think module_ptr should be used here because this function is called by module?